config/checkstyle/checkstyle.xml

```diff
@@ -1,22 +1,149 @@
 <?xml version="1.0"?>
 <!DOCTYPE module PUBLIC
         "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
-        "https://checkstyle.org/dtds/configuration_1_3.dtd" [
-        <!ENTITY checkerModuleDirectives SYSTEM "checker_module_directives.xml">
-        <!ENTITY prohibitJavaCollections SYSTEM "prohibit_java_collections.xml">
-        <!ENTITY treeWalkerModuleDirectives SYSTEM "tree_walker_module_directives.xml">
-        ]>
+        "https://checkstyle.org/dtds/configuration_1_3.dtd">
 
-<!-- List of directives: https://checkstyle.sourceforge.io/version/4.4/availablechecks.html -->
+<module name="Checker">
+    <module name="NewlineAtEndOfFile"/>
 
-<!-- DTD entities are used to allow for splitting config file to smaller parts
-     that can be reimported in submodule-specific configuration files. -->
+    <module name="SuppressionSingleFilter">
+        <property name="checks" value=".*"/>
+        <property name="files" value="^.*/build/generated/.*\.java$"/>
+    </module>
 
-<module name="Checker">
-    &checkerModuleDirectives;
+    <!-- to enable @SuppressWarnings -->
+    <module name="SuppressWarningsFilter"/>
 
     <module name="TreeWalker">
-        &prohibitJavaCollections;
-        &treeWalkerModuleDirectives;
+        <!-- to enable @SuppressWarnings -->
+        <module name="SuppressWarningsHolder"/>
+
+        <module name="ArrayTypeStyle"/>
+        <module name="AvoidStarImport"/>
+        <module name="ConstantName">
+            <property name="format" value="^((.+C|c)ache|(.+I|i)nstance|[A-Z][A-Z0-9]*(_[A-Z0-9]+)*)$"/>
+        </module>
+        <module name="CovariantEquals"/>
+        <module name="DefaultComesLast"/>
+        <module name="EmptyBlock">
+            <property name="option" value="TEXT"/>
+            <property name="tokens"
+                      value="LITERAL_TRY, LITERAL_FINALLY, LITERAL_IF, LITERAL_ELSE, LITERAL_SWITCH"/>
+        </module>
+        <module name="EmptyStatement"/>
+        <module name="EqualsHashCode"/>
+        <module name="FallThrough"/>
+        <module name="FinalClass"/>
+        <module name="HiddenField">
+            <property name="ignoreConstructorParameter" value="true"/>
+        </module>
+        <module name="HideUtilityClassConstructor"/>
+        <module name="IllegalCatch"/>
+        <module name="IllegalImport">
+            <!-- Avoid mutable collections; if really needed, qualify their class names explicitly. -->
+            <property name="illegalClasses" value="java.util.Deque, java.util.HashMap, java.util.List, java.util.Map, java.util.Queue, java.util.Set, java.util.Stack, java.util.TreeMap, java.util.Vector"/>
```

Comment:
do we need both this and `ForbiddenClassesTestSuite#no_classes_should_depend_on_java_collections`? :thinking:

Reply:
Yeah, coz we still want the Java collections to be explicitly imported where they're used (e.g. in `*.impl.*` packages)

Reply:
And this is by design **not** enforceable by ArchUnit since it only operates on classfiles, not source code... as opposed to Checkstyle, Lombok, and (for the most part) Checker Framework